### ISSUE TITLE
train error-ValueError: No variables provided.

### ISSUE THREAD

**Issue by DetectionIIT** [MarvinTeichmann/KittiSeg#166]
File "train.py", line 131, in <module>
    tf.app.run()
  File "/home/emma/anaconda3/envs/tensorflow/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 126, in run
    _sys.exit(main(argv))
  File "train.py", line 127, in main
    train.do_training(hypes)
  File "incl/tensorvision/train.py", line 377, in do_training
    tv_graph = core.build_training_graph(hypes, queue, modules)
  File "incl/tensorvision/core.py", line 100, in build_training_graph
    global_step, learning_rate)
  File "/home/emma/fcn/KittiSeg/hypes/../optimizer/generic_optimizer.py", line 91, in training
    global_step=global_step)
  File "/home/emma/anaconda3/envs/tensorflow/lib/python3.6/site-packages/tensorflow/python/training/optimizer.py", line 529, in apply_gradients
    raise ValueError("No variables provided.")
ValueError: No variables provided.

I don't know how to do with this,thanks with you help

**Comment by yuleaf612** [comment]
@emmra I met with the same error, have u solved it?

**Comment by zhaojingji** [comment]
It because the return format of zip() is different between python2 and 3.
change the line grads_and_vars = zip(clipped_grads, tvars) in optimizer/generic_optimizer.py to grads_and_vars = list(zip(clipped_grads, tvars)) @yuleaf612 @emmra

**Comment by yanchunhui** [comment]
@emmra @yuleaf612 I meet the same problem,too,so do you solved it?

**Comment by yuleaf612** [comment]
@yanchunhui yeah, I uninstalled everything. And I eventually intalled python 2.7 and tensorflow as it exactlly requried by this project

**Comment by yanchunhui** [comment]
@yuleaf612 thx for your reply. I also get the method like this, I will try it.

**Comment by yanchunhui** [comment]
amazing, I've found a solution to this problem from here, [https://stackoverflow.com/a/43538898](url)
now, it works very well.
